### PR TITLE
ci: update CI to publish android artifacts + disable temporarily android artifacts build (web3 not working)

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

- delete `java/` folder and move `build.gradle` under `scripts/`

### CODE REVIEW

**Review by skimaharvey — APPROVED** [review]

**Review by b00ste — APPROVED** [review]

**Comment by CJ42** [comment]
@Jenea keeping this PR as a draft until we manage to resolve publishing the android artifacts. on Maven registry.

**Comment by CJ42** [comment]
The CI will always fail as we have multiple `Transfer` events defined for LSP7CompatibleERC20 / LSP8CompatibleERC721

Maybe we should skip them for the time being in the Android artifacts.

https://github.com/web3j/web3j/issues/1781

File: .github/workflows/build-artifacts.yml [hunk 1]
```diff
@@ -28,35 +28,50 @@ jobs:
...
+      #     distribution: "adopt"
 
-      - name: Build jar
-        working-directory: ./java
-        run: gradle build
+      # - name: Build jar
+      #   working-directory: ./scripts/java
+      #   run: |
+      #     chmod +x build.gradle
+      #     gradle build
```
Comment (JeneaVranceanu) [inline .github/workflows/build-artifacts.yml:66]: 👍🏻

File: scripts/android-artifacts.ts [hunk 2]
```diff
@@ -2,23 +2,43 @@ const fs = require("fs");
 const hre = require("hardhat");
 const { exec } = require("child_process");
 
-const destination = "./java/src/main/java/network/lukso/up/contracts";
-const package = "network.lukso.up.contracts";
+/**
+ * @dev temporarily exclude the LSP7/8 Compatible contracts as these contain two `Transfer` and `Approve` events
+ * with the same name and different parameters. This causes Web3j to crash when building the java artifacts
+ */
```
Comment (Hugoo) [inline scripts/android-artifacts.ts:8]: 👍

**Review by Hugoo — APPROVED** [review]

File: .github/workflows/publish-android-artifacts.yml [hunk 3]
```diff
@@ -1,31 +1,44 @@
...
 
-      - name: Download artifact
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: "11"
+          distribution: "adopt"
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
```
[the remark this answers is not in the document]
Comment (CJ42) [reply .github/workflows/publish-android-artifacts.yml:26]: good point. It was taken from the example in the Github docs.

https://docs.github.com/en/actions/publishing-packages/publishing-java-packages-with-gradle#publishing-packages-to-github-packages

image

**Pull request merged**